### PR TITLE
[TSI] Fix response type for Create or Replace Types

### PR DESCRIPTION
**Pull request by bikamani**
Create or Replace Types returns Type/Error in return and earlier this API was only returning error object, fixing that and some description, tests accordingly in this PR

### CODE REVIEW

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 1]
```diff
@@ -1714,7 +1714,7 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
         /// <exception cref="ArgumentException">
         /// The exception is thrown when <paramref name="timeSeriesTypes"/> is empty.
         /// </exception>
-        public virtual async Task<Response<TimeSeriesOperationError[]>> CreateOrReplaceTimeSeriesTypesAsync(
+        public virtual async Task<Response<TimeSeriesTypeOperationResult[]>> CreateOrReplaceTimeSeriesTypesAsync(
```
Comment (bikamani) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1717]: returns a type object or error depending on success/failure of the create/replace operation.

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 2]
```diff
@@ -1741,7 +1741,7 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
                 // it will be placed at the same index location that corresponds to its place in the input array.
                 IEnumerable<TimeSeriesOperationError> errorResults = executeBatchResponse.Value.Put.Select((result) => result.Error);
```
Comment (barustum) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1742]: this line can now be removed

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 3]
```diff
@@ -1793,7 +1793,7 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
...
```
Comment (barustum) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1794]: this line can now be removed.

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 4]
```diff
@@ -1808,8 +1808,7 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
         /// <param name="timeSeriesTypeNames">List of names of the Time Series types to delete.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>
-        /// List of type or error objects corresponding by position to the array in the request.
-        /// Type object is set when operation is successful and error object is set when operation is unsuccessful.
+        /// List of error objects corresponding by position to the request array in the request - null when the operation is successful.
```
Comment (barustum) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1811]: maybe reword to "...to the input array in the request ...."

**Review by barustum — APPROVED** [review]
